init.go

```diff
@@ -0,0 +1,245 @@
+		},
+		cli.BoolFlag{
+			Name:  "refs, r",
+			Usage: "extract VCS refs for packages",
+		},
+	},
+}
+
+// depFinder finds the dependencies of an existing project by traversing go deps
+type depFinder struct {
+	// the seen dependencies during traversal
+	deps map[string]*pkg
+
+	// the root package. we ignore all its sub packages
+	root string
+
+	// a list of ignore import prefixes (e.g. gitlab.mydomain.com)
+	ignored []string
+
+	// the current working directory (this changes durgin traversal)
+	wd string
+
+	// output verbose messages during traversal
+	verbose bool
+
+	lister pkgLister
+}
+
+// isIgnored checks if an import path start with a prefix that has been set to be ignored
+func (c *depFinder) isIgnored(path string) bool {
+
+	for _, prefix := range c.ignored {
+		if strings.HasPrefix(path, prefix) {
+			return true
+		}
+	}
+
+	return false
+}
+
+// findDeps recursively finds all the dependencies of a package that are not part of the same
+// project or the standard library
+func (c *depFinder) findDeps(p *pkg) error {
+
+	if c.root == "" {
+		c.root = p.ImportPath
+	}
+
+	deps, err := c.lister.List(p.Deps...)
+	if err != nil {
+		return err
+	}
+
+	for _, dep := range deps {
+		if dep.Standard {
+			continue
+		}
+
+		// do not scan deps twice
+		if _, found := c.deps[dep.ImportPath]; found {
+			continue
+		}
+
+		if !strings.HasPrefix(dep.ImportPath, c.root) && !c.isIgnored(dep.ImportPath) {
+			if c.verbose {
+				fmt.Println(dep.ImportPath)
+			}
+			c.deps[dep.ImportPath] = dep
+
+		} else {
+			// we put ignored packages in the dep map as nils just so we'll scan them once for their deps
+			c.deps[dep.ImportPath] = nil
+		}
+
+		// we find deps even on "ignored" packages - we want their deps
+		c.findDeps(dep)
+	}
+
+	return nil
+}
+
+// findDepsInDir finds all the deps for the packages under the dir in path.
+// this is used to recursively traverse all packages in a project
+func (c *depFinder) findDepsInDir(pth string) error {
+
+	c.wd = pth
+	os.Chdir(c.wd)
+
+	pkgs, err := c.lister.List("")
+	if err != nil {
+		return err
+	}
+	for _, p := range pkgs {
+		c.findDeps(p)
+	}
+
+	files, err := ioutil.ReadDir(pth)
+
+	if err != nil {
+		return err
+	}
+
+	for _, file := range files {
+		if !file.IsDir() {
+			continue
+		}
+
+		if strings.HasPrefix(file.Name(), ".") || strings.HasPrefix(file.Name(), "_") {
+			continue
+		}
+
+		err := c.findDepsInDir(path.Join(pth, file.Name()))
+		if err != nil {
+			fmt.Println("error traversing", file.Name(), ": ", err)
+			return err
+		}
+
+	}
+
+	return nil
+
+}
+
+// FindDependencies traverses the current WD and retrieves a dependency map (package => revision)
+// for all 3rd party libraries.
+// VCS refs are retrieved only if the vcsRefs flag is set to true
+func (c *depFinder) FindDependencies(vcsRefs bool) (map[string]string, error) {
+	err := c.findDepsInDir(c.wd)
+	if err != nil {
+		return nil, err
+	}
+
+	ret := make(map[string]string)
+	if c.verbose && vcsRefs {
```

Comment:
I'm a bit confused on `vcsRefs`. Are we trying to get the SHA from dependencies in the `GOPATH`?